src/test/java/bio/terra/integration/TestJobWatcher.java

```diff
@@ -0,0 +1,54 @@
+package bio.terra.integration;
+
+import bio.terra.common.configuration.TestConfiguration;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.Instant;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.stringtemplate.v4.ST;
+
+// This class can be used in a rule for integration test to decorate failures to make it easier to
+// debug
+public class TestJobWatcher extends TestWatcher {
```

Comment:
This is an interesting idea, it looks like it will be a big help getting to errors in integration tests. It's too bad you can't inject directly into it though.

According to stackoverflow you can make a rule injectable (which makes sense, since spring DI can work in any object). https://stackoverflow.com/a/56343480

Reply:
interesting...though I guess injecting vs. just instantiating isn't a huge difference

Reply:
The difference is you don't have to declare this class's required injections in the calling class. The caller shouldn't need to know that this object needs `TestConfiguration` to work.